src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -1131,6 +1131,9 @@ public void MoveToRecycleBin(IContent content, int userId = 0)
                     return;
                 }
 
+                // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                ApplicationContext.Current.Services.RelationService.Relate(content.Parent(), content, Constants.Conventions.Relations.RelateParentDocumentOnDelete);

```

Comment:
:+1: 

Let me know if you have questions around the implementation of this. Otherwise I'll keep an eye out for your update for review :smiley: